ecosystem/node-checker/src/metric_evaluator/traits.rs

```diff
@@ -0,0 +1,64 @@
+// Copyright (c) Aptos
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::public_types::Evaluation;
+use anyhow::{Error, Result};
+use prometheus_parse::Scrape as PrometheusScrape;
+use std::fmt::{Display, Formatter, Result as FmtResult};
+
+// TODO: Consider using thiserror.
+
+#[derive(Debug)]
+pub enum MetricsEvaluatorError {
+    /// The metric we're evaluating is missing from the baseline. Args:
+    ///   - The metric name.
+    ///   - Explanation.
+    /// When the target node is missing a metric, we return an Evaluation
+    /// indiating that something is wrong with the target node, but if the
+    /// baseline node is missing a metric, it implies that something is wrong
+    /// without our node checker configuration, so we return an error here.
+    MissingBaselineMetric(String, String),
+
+    UnknownError(Error),
+}
+
+impl Display for MetricsEvaluatorError {
```

Comment:
Will deriving `Display` not do the trick?

Reply:
Display cannot be derived sadly.

Reply:
Though thiserror impls it, which I just added 😄